app/controllers/admin/groups_controller.rb

```diff
@@ -24,13 +24,22 @@ def bulk
 
   def bulk_perform
     group = Group.find(params[:group_id].to_i)
+    users_added = 0
     if group.present?
       users = (params[:users] || []).map {|u| u.downcase}
-      user_ids = User.where("username_lower in (:users) OR email IN (:users)", users: users).pluck(:id)
-      group.bulk_add(user_ids) if user_ids.present?
+      valid_emails = valid_usernames = {}
```

Comment:
Both of these variables are actually referring to the same hash so I'm not sure if there is what you intended?